src/convertToObject.js

```diff
@@ -7,6 +7,20 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  const result = {};
+
+  const x = sourceString
+    .toString()
+    .split(';')
+    .map((item) => item.toString().split(':'));
+
+  x.map((item) => {
+    if (!(item[1] === undefined)) {
+      result[item[0].trim()] = item[1].trim();
+    }
```

Comment:
[CODE PRACTICE] - Avoid using loops in this task, use iteration methods instead. Although you are using `.map`, it is being used incorrectly. Consider using `.reduce` to directly build the resulting object without the need for an external variable or loop.